mesonbuild/mesonlib.py

```diff
@@ -409,14 +410,59 @@ def make_same_len(listA, listB):
         for n in range(len(i), maxlen):
             i.append(0)
 
-numpart = re.compile('[0-9.]+')
+# a helper class which implements the same version ordering as RPM
+@functools.total_ordering
+class Version:
+    def __init__(self, s):
+        self._s = s
 
-def version_compare(vstr1, vstr2, strict=False):
-    match = numpart.match(vstr1.strip())
-    if match is None:
-        msg = 'Uncomparable version string {!r}.'
-        raise MesonException(msg.format(vstr1))
-    vstr1 = match.group(0)
+        # split into numeric, alphabetic and non-alphanumeric sequences
+        sequences = re.finditer(r'(\d+|[a-zA-Z]+|[^a-zA-Z\d]+)', s)
+        # non-alphanumeric separators are discarded
+        sequences = [m for m in sequences if not re.match(r'[^a-zA-Z\d]+', m.group(1))]
+        # numeric sequences have leading zeroes discarded
+        sequences = [re.sub(r'^0+(\d)', r'\1', m.group(1), 1) for m in sequences]
```

Comment:
A more readable version of same is `txt.lstrip('0')`. This does not need to be fixed to get merged, just something to keep in mind in the future.

Reply:
Good point.

But it turns out that's not quite the same.  When applied to '0', '00', '000' etc., the subsitution gives '0', whereas lstrip gives ''.